.github/workflows/build.yml

```diff
@@ -105,7 +101,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [ 17, 11 ]  # If we change these, make sure to adjust ci-complete.yml
+        java: [ 21, 11 ]  # If we change these, make sure to adjust ci-complete.yml
```

Comment:
I have left a comment on https://issues.apache.org/jira/browse/KAFKA-17684, but this is a better place to discuss it :smiley: 

KAFKA-16096 want to raise the minimum java version to 17 for broker, so IMHO, the tests should be run on 17 and 21. Otherwise, the CI run on 11 can't build the modules have `release=17`

Reply:
If we have clients/streams/connect compiling with release=11, should we not also run the JUnit tests on 11? 

To achieve the goals of the KIP, I wonder if we'll need to split the build into client components (release=11, test 11 and 21) and server components (release=17, test 17 and 21)

Reply:
> I wonder if we'll need to split the build into client components (release=11, test 11 and 21) and server components (release=17, test 17 and 21)

I assumed the golden rule is to minimize the usage of GitHub runner :smile: 

What about using `release=11 test 11`, `release=11 test 17`, and `release=11/17 build 21`? I believe bugs caused by different JDK versions are rare, and we can run tests for all supported JDKs when preparing a release.

Reply:
> I assumed the golden rule is to minimize the usage of GitHub runner

💯 

I think that combination of versions makes sense. Once we have updated Gradle to split the `--release` versions between the different modules, we can update the CI to test different targets. 

For now, since we are running all modules together in one build, we can run with the oldest and newest across those versions (which is 11 and 21). WDYT?